google/resource_container_cluster.go

```diff
@@ -844,6 +907,20 @@ func flattenClusterNodePools(d *schema.ResourceData, config *Config, c []*contai
 	return nodePools, nil
 }
 
+func flattenMasterAuthorizedNetworksConfig(c *container.MasterAuthorizedNetworksConfig) []map[string]interface{} {
+	cidrBlocks := make([]string, 0, len(c.CidrBlocks))
+	for _, v := range c.CidrBlocks {
+		cidrBlocks = append(cidrBlocks, v.CidrBlock)
+	}
+	result := []map[string]interface{}{
+		map[string]interface{}{
+			"enabled":     c.Enabled,
+			"cidr_blocks": cidrBlocks,
```

Comment:
Turns out, if `flattenMasterAuthorizedNetworksConfig` returns something that looks like this:
```
result: [map[enabled:false cidr_blocks:[]]]
```
`d.Set` will actually return an error: `Invalid address to set: []string{"master_authorized_networks_config", "0", "cidr_blocks"}` This is fairly cryptic, but it basically just means that you can't set an empty slice in config, you have to just set no slice at all.

Reply:
Hrmmmmm, alright.  I'll add that as a test case, too.

Reply:
Done.